packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts

```diff
@@ -80,40 +95,38 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
 
   // TODO: Improve this uninstallation cleanup per event by keeping a relation of EventType to App which has the data.
   for (const eventType of eventTypes) {
-    if (eventType.locations) {
-      // If it's a video, replace the location with Cal video
-      if (
-        credential.app?.categories.includes(AppCategories.video) ||
-        credential.app?.categories.includes(AppCategories.conferencing)
-      ) {
-        // Find the user's event types
-
-        // Look for integration name from app slug
-        const integrationQuery =
-          credential.app?.slug === "msteams" ? "office365_video" : credential.app?.slug.split("-")[0];
-
-        // Check if the event type uses the deleted integration
-
-        // To avoid type errors, need to stringify and parse JSON to use array methods
-        const locationsSchema = z.array(z.object({ type: z.string() }));
-        const locations = locationsSchema.parse(eventType.locations);
-
-        const updatedLocations = locations.map((location: { type: string }) => {
-          if (location.type.includes(integrationQuery)) {
-            return { type: DailyLocationType };
-          }
-          return location;
-        });
+    // If it's a video, replace the location with Cal video
+    if (eventType.locations && isVideoOrConferencingApp(credential.app)) {
+      // Find the user's event types
 
-        await prisma.eventType.update({
-          where: {
-            id: eventType.id,
-          },
-          data: {
-            locations: updatedLocations,
-          },
-        });
-      }
+      const integrationQuery = getRemovedIntegrationNameFromAppSlug(credential.app?.slug ?? "");
+
+      // Check if the event type uses the deleted integration
+
+      // To avoid type errors, need to stringify and parse JSON to use array methods
+      const locations = locationsSchema.parse(eventType.locations);
+
+      const doesDailyVideoAlreadyExists = locations.some((location) =>
+        location.type.includes(DailyLocationType)
+      );
+
+      const updatedLocations: TlocationsSchema = locations.reduce((acc: TlocationsSchema, location) => {
+        if (location.type.includes(integrationQuery)) {
+          if (!doesDailyVideoAlreadyExists) acc.push({ type: DailyLocationType });
+        } else {
+          acc.push(location);
+        }
+        return acc;
+      }, []);
+
+      await prisma.eventType.update({
+        where: {
+          id: eventType.id,
+        },
+        data: {
+          locations: updatedLocations,
+        },
+      });
```

Comment:
You code worked fine but it wasn't easily readable/maintainable so i pushed these changes.

Reply:
Thank you!!!!, looks good 👍